bin/helpers/utils.js

```diff
@@ -116,6 +116,9 @@ exports.setParallels = (bsConfig, args) => {
 };
 
 exports.setUsername = (bsConfig, args) => {
+  if (this.isUndefined(bsConfig['auth']) && (!this.isUndefined(args.username) || !this.isUndefined(process.env.BROWSERSTACK_USERNAME))) {
```

Comment:
We check and set it in validateBstackJson function instead since it is what loads the file and sents out the JSON back to all modules.

Reply:
validateBstackJson function just checks if it is a validate json or not . After that we set this username and other variables set by using environment variables and post which the validations are performed.

Reply:
Ok.

Why do we need an extra checks after && ? Why can't we just set the auth object if it is missing directly? Also, why not do it in setAccessKey?

Should we create another function in utils just for this that we use before setting username and access key? Thoughts?

Reply:
If we don't have extra checks then , if the user doesn't pass any args or doesn't set via env vars the existing validation breaks and throws code error instead of validation error. 
We can create another function but would be an overhead as we already have this function .

Reply:
>  We can create another function but would be an overhead as we already have this function .
Let's create another. Seem like a check/validation we should do separately and not involding username/accesskey domains.